astropy/io/misc/hdf5.py

```diff
@@ -276,6 +311,21 @@ def write_table_hdf5(table, output, path=None, compression=False,
         else:
             raise OSError("Table {0} already exists".format(path))
 
+    # Encode any mixin columns as plain columns + appropriate metadata
+    table = _encode_mixins(table)
+
+    # Warn if information will be lost when serialize_meta=False.  This is
+    # hardcoded to the set difference between column info attributes and what
+    # HDF5 can store natively (name, dtype) with no meta.
+    if serialize_meta is False:
```

Comment:
Note: this warning is technically unrelated to mixin support but in the spirit of #6912 it seems like a good thing.